services/docker/docker_test.go

```diff
@@ -0,0 +1,102 @@
+package docker
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/honeytrap/honeytrap/pushers"
+)
+
+type Test struct {
+	Name           string
+	ReqMethod      string
+	ReqPath        string
+	ReqBody        string
+	ExpectedBody   string
+	ExpectedStatus int
+}
+
+var tests = []Test{
+	{
+		Name:           "version_root",
+		ReqMethod:      "GET",
+		ReqPath:        "/version",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "version",
+		ReqMethod:      "GET",
+		ReqPath:        "/1.41/version",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "info_root",
+		ReqMethod:      "GET",
+		ReqPath:        "/info",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "info",
+		ReqMethod:      "GET",
+		ReqPath:        "/1.41/info",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+}
+
+func TestDocker(t *testing.T) {
+	//Create Servicer
+	s := Docker()
+
+	// Create channel
+	s.SetChannel(pushers.MustDummy())
+
+	for _, tst := range tests {
+
+		//Create a pipe
+		client, server := net.Pipe()
+		defer client.Close()
+		defer server.Close()
+
+		go s.Handle(context.TODO(), server)
+
+		req := httptest.NewRequest(tst.ReqMethod, tst.ReqPath, strings.NewReader(tst.ReqBody))
+		if err := req.Write(client); err != nil {
+			t.Error(err)
```

Comment:
nit: you probably want t.Fatal so that it doesn't continue on error?

but also, we generally just use https://godoc.org/github.com/stretchr/testify/require

Reply:
:+1: will avoid adding libs for now

Reply:
I checked the dependencies (I think) - the repo already depends on those libs, so you should be safe to use them.